WireDomain/Sources/WireDomain/Repositories/ConversationsLabels/ConversationLabelsRepository.swift

```diff
@@ -119,27 +172,33 @@ final class ConversationLabelsRepository: ConversationLabelsRepositoryProtocol {
 
             deleteRequest.resultType = .resultTypeObjectIDs
 
-            let batchDelete = try context.execute(deleteRequest) as? NSBatchDeleteResult
+            do {
+                let batchDelete = try context.execute(deleteRequest) as? NSBatchDeleteResult
 
-            guard let deleteResult = batchDelete?.result as? [NSManagedObjectID] else {
-                throw ConversationLabelsRepositoryError.failedToDeleteStoredLabels
-            }
+                guard let deleteResult = batchDelete?.result as? [NSManagedObjectID] else {
+                    throw ConversationLabelsRepositoryError.failedToDeleteStoredLabels
+                }
 
-            let deletedObjects: [AnyHashable: Any] = [
-                NSDeletedObjectsKey: deleteResult
-            ]
+                let deletedObjects: [AnyHashable: Any] = [
+                    NSDeletedObjectsKey: deleteResult
+                ]
 
-            /// Since `NSBatchDeleteRequest` only operates at the SQL level (in the persistent store itself),
-            /// we need to manually update our in-memory objects after execution.
+                /// Since `NSBatchDeleteRequest` only operates at the SQL level (in the persistent store itself),
+                /// we need to manually update our in-memory objects after execution.
 
-            NSManagedObjectContext.mergeChanges(
-                fromRemoteContextSave: deletedObjects,
-                into: [context]
-            )
+                NSManagedObjectContext.mergeChanges(
+                    fromRemoteContextSave: deletedObjects,
+                    into: [context]
+                )
+
+                /// Ensures the context and the persistent store are in sync
 
-            /// Ensures the context and the persistent store are in sync
+                try context.save()
```

Comment:
if this save fails, it does not mean we failed to delete the labels but just failed to purge in memory objects

Reply:
I think calling `try context.save()` is irrelevant here since we already manually update our in-memory objects with the following method. I'll remove it.

```
NSManagedObjectContext.mergeChanges(
    fromRemoteContextSave: deletedObjects,
    into: [context]
)

```